synchronization/block_tracker.go

```diff
@@ -48,21 +48,21 @@ func (t *BlockTracker) readAtomicHeightAndLatch() (uint64, chan struct{}) {
 
 func (t *BlockTracker) WaitForBlock(requestedHeight primitives.BlockHeight) error {
 
-	requestedHeightUint := uint64(requestedHeight)
+	requestedHeightInt := int64(requestedHeight)
```

Comment:
This will break when requestedHight is larger than the max value of int64. suggest to avoid this cast by using addition instead of subtraction below in line 58.